src/EnergyPlus/GroundTemperatureModeling/FiniteDifferenceGroundTemperatureModel.hh

```diff
@@ -0,0 +1,199 @@
+#ifndef FiniteDifferenceGroundTemperatureModel_hh_INCLUDED
+#define FiniteDifferenceGroundTemperatureModel_hh_INCLUDED
+
+// C++ Headers
+#include <memory>
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/Array1D.hh>
+#include <ObjexxFCL/Array2D.hh>
+#include <ObjexxFCL/Optional.hh>
+
+// EnergyPlus Headers
+#include <EnergyPlus.hh>
+#include <GroundTemperatureModeling/BaseGroundTemperatureModel.hh>
+
+namespace EnergyPlus {
+
+	// Derived class for Finite-Difference Model
+	class FiniteDiffGroundTempsModel : public BaseGroundTempsModel {
+		
+		public:
+			Real64 baseConductivity;
+			Real64 baseDensity;
+			Real64 baseSpecificHeat;
+			int totalNumCells;
+			Real64 timeStepInSeconds;
+			Real64 evapotransCoeff;
+			Real64 saturatedWaterContent; 
+			Real64 waterContent;
+			Real64 annualAveAirTemp;
+			Real64 minDailyAirTemp; // Set hi. Will be reset later
+			Real64 maxDailyAirTemp; // Set low. Will be reset later
+			Real64 dayOfMinDailyAirTemp;
+			Real64 depth;
+			Real64 simTimeInDays;
+
+		// Default constructor
+		FiniteDiffGroundTempsModel() :
```

Comment:
@mmatts Since you are using a factory method to create these objects, this should probably be private instead of public. Also does all the data in the class need to be public or can some of it be private? Are any of the member variables common among all the derived classes? If so, they could move to the base class and be 'protected' visibility.